### PR TITLE
[Linux] Fix tier used in debian watch files

### LINKED ISSUE #2552: [Linux] Update Debian watch files to specify TIER
Currently, the Debian watch files follow a pattern
```
https://downloads.keyman.com/linux/(?:beta|stable)/(\d\S+)/keyman_config-(.+)\.tar\.gz debian uupdate
```

But the uscan is only getting the latest `beta` release. This hampers `stable` releases from shipping to launchpad. From the uscan [documentation](https://manpages.debian.org/stretch/devscripts/uscan.1.en.html)
> (If multiple directories match, the highest version is picked.

### Affected files
/common/engine/keyboardprocessor/debian/watch
/linux/**/debian/watch

Need to investigate if `${TIER}` can be applied in the watch files...

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #2552 

The current debian/watch files won't work for `stable` tier because the regex `(beta|stable)` matches the existing beta versions first.

Also, to support upcoming 14.0, the tier will be set by a `TIER.md` file.

This PR updates launchpad.sh so that:
1. tier is determined by `TIER.md` file or environment var $TIER. Fail if not set
2. Use `sed` to update debian/watch files with the specified `${tier}`. This will overwrite the tier from a previous run.

### CODE REVIEW

File: common/engine/keyboardprocessor/debian/watch [hunk 1]
```diff
@@ -1,2 +1,2 @@
 version=4
-https://downloads.keyman.com/linux/(?:beta|stable)/(\d\S+)/keyman-keyboardprocessor-(.+)\.tar\.gz debian uupdate
+https://downloads.keyman.com/linux/(?:$tier)/(\d\S+)/keyman-keyboardprocessor-(.+)\.tar\.gz debian uupdate
```
Comment (ermshiperete) [inline common/engine/keyboardprocessor/debian/watch]: > .../(?:$tier)/...

Does this work? I didn't know you can use variables here. Oh, I see, it gets replaced by `launchpad.sh` (I wonder if a comment would be helpful?)

There's probably no need to use a regex now. Just `.../$tier/...` should work as well I'd think.

**Comment by mcdurdin** [comment]
Note: jenkins build fails because it is expecting TIER.md to exist (and corresponding resources/build files, and root level VERSION.md to actually contain the version), all of which is only on master branch at present.

File: common/engine/keyboardprocessor/debian/watch (hunk 1 above)
Comment (darcywong00) [reply common/engine/keyboardprocessor/debian/watch]: k, I've added a comment to the watch files and removed the regex

File: linux/ibus-keyman/debian/control [hunk 2]
```diff
@@ -4,7 +4,7 @@ Priority: optional
 Maintainer: Daniel Glassey <wdg@debian.org>
 Build-Depends: debhelper (>= 9), autotools-dev, dh-autoreconf, pkg-config,
  libkmnkbp-dev (>=0.0.0~), libibus-1.0-dev (>= 1.2), libjson-glib-dev (>= 1.4.0) <!pkg.keyman-config.xenial>,
- libjson-glib-dev (1.1.2) <pkg.keyman-config.xenial>, libgtk-3-dev
+ libjson-glib-dev (= 1.1.2) <pkg.keyman-config.xenial>, libgtk-3-dev
```
Comment (darcywong00) [inline linux/ibus-keyman/debian/control]: Missed this in #2648

**Pull request merged**